proto/dydxprotocol/govplus/tx.proto

```diff
@@ -1,7 +1,58 @@
 syntax = "proto3";
 package dydxprotocol.govplus;
 
+import "cosmos_proto/cosmos.proto";
+import "cosmos/msg/v1/msg.proto";
+import "gogoproto/gogo.proto";
+import "amino/amino.proto";
+
 option go_package = "github.com/dydxprotocol/v4-chain/protocol/x/govplus/types";
 
 // Msg defines the Msg service.
-service Msg {}
+service Msg {
+  // SlashValidator is exposed to allow slashing of a misbehaving validator via
+  // governance.
+  rpc SlashValidator(MsgSlashValidator) returns (MsgSlashValidatorResponse);
+}
+
+// MsgSlashValidator is the Msg/SlashValidator request type.
+message MsgSlashValidator {
+  // The address that controls the module (likely the gov module account).
+  option (cosmos.msg.v1.signer) = "authority";
+  string authority = 1;
+
+  // Address of the validator to slash
+  string validator_address = 2;
+
+  // Colloquially, the height at which the validator is deemed to have
+  // misbehaved. In practice, this is the height used to determine the targets
+  // of the slash. For example, undelegating after this height will not escape
+  // slashing. This height should be set to a recent height at the time of the
+  // proposal to prevent delegators from undelegating during the vote period.
+  //
+  // NB: At the time this message is applied, this height must have occured
+  // equal to or less than an unbonding period in the past in order for the
+  // slash to be effective.
+  uint32 infraction_height = 3;
+
+  // Power of the validator at the specified height. Used to compute the slash
+  // amount. Note that this is not directly the token amount; it must be
+  // adjusted by power reduction to get consensus power.
+  bytes power_at_infraction_height = 4 [
+    (gogoproto.customtype) =
+        "github.com/dydxprotocol/v4-chain/protocol/dtypes.SerializableInt",
+    (gogoproto.nullable) = false
+  ];
+
+  // Multiplier for how much of the validator's stake should be slashed.
+  // slash_factor * power = amount slashed
+  string slash_factor = 5 [
```

Comment:
For my understanding why is the extra options needed, besides `[(cosmos_proto.scalar) = "cosmos.Dec"]` ? The gov v1 [fields](https://github.com/dydxprotocol/cosmos-sdk/blob/bdf96fddf421e90574e53af383c1468fdb33d7a3/proto/cosmos/gov/v1/gov.proto#L174C21-L174C59) only have this one option, while v1beta fields have more options ([example](https://github.com/dydxprotocol/cosmos-sdk/blob/bdf96fddf421e90574e53af383c1468fdb33d7a3/proto/cosmos/staking/v1beta1/staking.proto#L34-L37))

Reply:
cosmos_protos are just annotations. gogoproto.customtype adds some codegen to help using that type. gogoproto.nullable affects the codegen (pointer or value)